pkg/azuredisk/nodeserver.go

```diff
@@ -296,14 +296,27 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 		defer cancel()
 		result, err := d.cloud.VirtualMachineSizesClient.List(ctx, d.cloud.Location)
 		if err != nil || result.Value == nil {
-			return nil, status.Error(codes.Internal, "failed to list vm sizes in GetVolumeLimits from Azure cloud provider, nodeName: "+d.NodeID)
+			klog.Warningf("list vm sizes of nodeName(%s) failed with error: %v ", d.NodeID, err)
+		} else {
+			vmSizeList = result.Value
 		}
-		vmSizeList = result.Value
+	}
+
+	topology := &csi.Topology{
+		Segments: map[string]string{},
+	}
+	zone, err := d.cloud.GetZone(ctx)
+	if err != nil {
+		klog.Warningf("Failed to get zone from Azure cloud provider, nodeName: %v, error: %v", d.NodeID, err)
+	} else {
```

Comment:
I am running the tests on a cluster with not availiability zone and the E2E used the following storage class:
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2019-11-05T18:40:13Z"
  generateName: azuredisk-613-disk.csi.azure.com-dynamic-sc-
  name: azuredisk-613-disk.csi.azure.com-dynamic-sc-jj7tz
  resourceVersion: "2038"
  selfLink: /apis/storage.k8s.io/v1/storageclasses/azuredisk-613-disk.csi.azure.com-dynamic-sc-jj7tz
  uid: 1a761ffe-0410-44eb-b466-145113153d26
parameters:
  skuName: Standard_LRS
provisioner: disk.csi.azure.com
reclaimPolicy: Delete
volumeBindingMode: Immediate
```
Here is a more detailed logs regarding the failure:
```log
GRPC request: name:"pvc-86b3ab84-01ce-4929-b98c-9b00cb566f60" capacity_range:<required_bytes:10737418240 > volume_capabilities:<mount:<fs_type:"ext4" > access_mode:<mode:SINGLE_NODE_WRITER > > parameters:<key:"skuName" value:"Standard_LRS" > accessibility_requirements:<requisite:<segments:<key:"topology.disk.csi.azure.com/zone" value:"0" > > preferred:<segments:<key:"topology.disk.csi.azure.com/zone" value:"0" > > >
I1105 18:40:28.469773       1 controllerserver.go:173] begin to create azure disk(pvc-86b3ab84-01ce-4929-b98c-9b00cb566f60) account type(Standard_LRS) rg(chuwon) location() size(10)
I1105 18:40:28.469801       1 azuredisk.go:246] azureDisk - creating new managed Name:pvc-86b3ab84-01ce-4929-b98c-9b00cb566f60 StorageAccountType:Standard_LRS Size:10
I1105 18:40:28.469829       1 azuredisk.go:252] azureDisk - AvailabilityZone: 0, zone list: []
E1105 18:40:28.518835       1 utils.go:117] GRPC error: compute.DisksClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidAvailabilityZone" Message="The zone(s) '' for resource 'Microsoft.Compute/disks/pvc-86b3ab84-01ce-4929-b98c-9b00cb566f60' is not supported. The supported zones for location 'westus2' are '1,2,3'"
```
From csi-azuredisk-node, you can see that 
```
I1105 18:36:33.291633       1 nodeserver.go:313] NodeGetInfo, nodeName: k8s-agentpool1-95854893-0, zone: 0
```
But the cluster is not using any AZ.

Reply:
The e2e test didn't catch the error because the topology feature gate wasn't enabled in the latest helm chart template. The e2e test uses the latest helm chart to install the csi driver before running the test.

Reply:
@chewong thanks, I have updated the chart config: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/176/commits/701db3bb330e9d8e91675c52741dec535982f058